pkg/splunk/enterprise/util.go

```diff
@@ -240,22 +244,31 @@ func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.Me
 	var localAppsConf, clusterAppsConf string
 	localAppsConf = yamlConfHeader
 	clusterAppsConf = yamlConfHeader
+	var mapKeys []string
 
-	for appSrc, appSrcDeployInfo := range appsSrcDeployStatus {
-		appDeployList := appSrcDeployInfo.AppDeploymentInfoList
+	// Map order is not guaranteed, so use the sorted keys to go through the map entries
+	for appSrc := range appsSrcDeployStatus {
+		mapKeys = append(mapKeys, appSrc)
+	}
+	sort.Strings(mapKeys)
+
+	for _, appSrc := range mapKeys {
+		appDeployList := appsSrcDeployStatus[appSrc].AppDeploymentInfoList
 
 		switch scope := getAppSrcScope(appConf, appSrc); scope {
 		case "local":
 			for idx := range appDeployList {
-				if appDeployList[idx].DeployStatus == enterprisev1.DeployStatusPending {
+				if appDeployList[idx].DeployStatus == enterprisev1.DeployStatusPending &&
```

Comment:
When does DeployStatus be set to Pending. What happens to configMap (& subsequent steps) contents if DeployStatus is not Pending?

Reply:
I'll let @sgontla answer that :)

Reply:
Had a chat with @sgontla 

 - `DeployStatus` will be set to Pending when there are App updates in `handleAppRepoChanges` (prior to call to `ApplyAppListingConfigMap`
 - If there are no updates to apps, configMap will have `app-list-local.yaml` & `app-list-cluster.yaml` maps with only headers (& empty data) which will avoid any unnecessary resets